### Context :

 A chit fund has 25 members. Each month they contribute Rs 2000 each. End of the month, theperson who bids the lowest for the corpus, gets his bid. The group organizer gets paid a fixedcommission of Rs 2500 each month (5% of 25*2000).It is deducted from the bid winner'scorpus. The remaining amount is distributed among all the 25 customers equally.Enclosedexcel has the table with cashflows for each month.

Task :
    1. What is the Annualized Return of the person who bids in the last month ?
    2. What is the Annualized Return of the person who bids in the first month 
    3. Write an R/Python script which calculates the annualized return of chit fund participant ?-

In [1]:
Chit_Fund_Montly_Port=25*2000

In [2]:
Chit_Fund_Montly_Port

50000

In [3]:
#Basic Libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

#Data Visualization Libraries
import seaborn as sns
import matplotlib.pyplot as plt

In [4]:
#Loading the dataset
df = pd.read_excel('Exercise Data.xlsx')
df

Unnamed: 0    Unnamed: 1                Unnamed: 2  \
0       Month  Contribution  Amount won by the bidder   
1           1          2000                     40000   
2           2          2000                     42000   
3           3          2000                     45000   
4           4          2000                     48000   
5           5          2000                     40000   
6           6          2000                     42000   
7           7          2000                     43000   
8           8          2000                     44000   
9           9          2000                     40000   
10         10          2000                     42000   
11         11          2000                     45000   
12         12          2000                     48000   
13         13          2000                     40000   
14         14          2000                     42000   
15         15          2000                     43000   
16         16          2000                     43000   
17         17          2000                     41000   
18         18          2000                     42000   
19         19          2000                     45000   
20         20          2000                     45000   
21         21          2000                     44000   
22         22          2000                     46000   
23         23          2000                     47000   
24         24          2000                     48000   
25         25          2000                     50000   

                        Unnamed: 3                     Unnamed: 4  \
0   Chit fund organizer commission  Net amount recd by Bid winner   
1                             2500                          37500   
2                             2500                          39500   
3                             2500                          42500   
4                             2500                          45500   
5                             2500                          37500   
6                             2500                          39500   
7                             2500                          40500   
8                             2500                          41500   
9                             2500                          37500   
10                            2500                          39500   
11                            2500                          42500   
12                            2500                          45500   
13                            2500                          37500   
14                            2500                          39500   
15                            2500                          40500   
16                            2500                          40500   
17                            2500                          38500   
18                            2500                          39500   
19                            2500                          42500   
20                            2500                          42500   
21                            2500                          41500   
22                            2500                          43500   
23                            2500                          44500   
24                            2500                          45500   
25                            2500                          47500   

                                  Unnamed: 5  
0   Amount returned to everyone in the group  
1                                        400  
2                                        320  
3                                        200  
4                                         80  
5                                        400  
6                                        320  
7                                        280  
8                                        240  
9                                        400  
10                                       320  
11                                 

In [5]:
#Change the colums name 
df.rename(columns=df.iloc[0], inplace=True)

In [6]:
df.head()

Month  Contribution  Amount won by the bidder  \
0  Month  Contribution  Amount won by the bidder   
1      1          2000                     40000   
2      2          2000                     42000   
3      3          2000                     45000   
4      4          2000                     48000   

   Chit fund organizer commission  Net amount recd by Bid winner  \
0  Chit fund organizer commission  Net amount recd by Bid winner   
1                            2500                          37500   
2                            2500                          39500   
3                            2500                          42500   
4                            2500                          45500   

   Amount returned to everyone in the group  
0  Amount returned to everyone in the group  
1                                       400  
2                                       320  
3                                       200  
4                                        80

In [7]:
#Drop index 0
df = df.drop(index=0)

In [8]:
df.head()

Month Contribution Amount won by the bidder Chit fund organizer commission  \
1     1         2000                    40000                           2500   
2     2         2000                    42000                           2500   
3     3         2000                    45000                           2500   
4     4         2000                    48000                           2500   
5     5         2000                    40000                           2500   

  Net amount recd by Bid winner Amount returned to everyone in the group  
1                         37500                                      400  
2                         39500                                      320  
3                         42500                                      200  
4                         45500                                       80  
5                         37500                                      400

In [9]:
df.columns

Index(['Month', 'Contribution', 'Amount won by the bidder',
       'Chit fund organizer commission', 'Net amount recd by Bid winner',
       'Amount returned to everyone in the group'],
      dtype='object')

In [10]:
#Toatal amount retured to everyone find out
df['Amount returned to everyone in the group'].sum()

6200

### Task1. What is the Annualized Return of the person who bids in the last month ?

In [11]:
df.tail()

Month Contribution Amount won by the bidder Chit fund organizer commission  \
21    21         2000                    44000                           2500   
22    22         2000                    46000                           2500   
23    23         2000                    47000                           2500   
24    24         2000                    48000                           2500   
25    25         2000                    50000                           2500   

   Net amount recd by Bid winner Amount returned to everyone in the group  
21                         41500                                      240  
22                         43500                                      160  
23                         44500                                      120  
24                         45500                                       80  
25                         47500                                        0

In [12]:
#Using Simple intrest formul
#I=P*r*t/100

In [13]:
#Toatl amount collect after mature chit fund last month
A25=47500+6200

In [14]:
A25

53700

In [15]:
#Toal principl amount of the last person
P25=50000

In [16]:
I25=A25-P25

In [17]:
#Total profit gain by last person
I25

3700

In [18]:
t25=25/12

In [19]:
r25=I25*100/(P25*t25)

In [20]:
r25

3.552

Ans1. Annualized Return of the person who bids in the last month=3.552

### Task2: What is the Annualized Return of the person who bids in the first month 

In [21]:
df.head()

Month Contribution Amount won by the bidder Chit fund organizer commission  \
1     1         2000                    40000                           2500   
2     2         2000                    42000                           2500   
3     3         2000                    45000                           2500   
4     4         2000                    48000                           2500   
5     5         2000                    40000                           2500   

  Net amount recd by Bid winner Amount returned to everyone in the group  
1                         37500                                      400  
2                         39500                                      320  
3                         42500                                      200  
4                         45500                                       80  
5                         37500                                      400

In [22]:
#Toal amount collect by first person after mature chit fund
A1=37500+6200

In [23]:
A1

43700

In [24]:
#Toal amount invest in chit fund by first prsonn
P1=50000

In [25]:
I1=A1-P1

In [26]:
I1

-6300

Find: 6300 rs lass by first person

In [27]:
t1=25/12

In [28]:
r1=I1*100/(P1*t1)

In [29]:
r1

-6.048

Ans2. Annualized Return of the person who bids in the first month= -6.048

### Task3:  Write an R/Python script which calculates the annualized return of chit fund participant ?

In [30]:
df

Month Contribution Amount won by the bidder Chit fund organizer commission  \
1      1         2000                    40000                           2500   
2      2         2000                    42000                           2500   
3      3         2000                    45000                           2500   
4      4         2000                    48000                           2500   
5      5         2000                    40000                           2500   
6      6         2000                    42000                           2500   
7      7         2000                    43000                           2500   
8      8         2000                    44000                           2500   
9      9         2000                    40000                           2500   
10    10         2000                    42000                           2500   
11    11         2000                    45000                           2500   
12    12         2000                    48000                           2500   
13    13         2000                    40000                           2500   
14    14         2000                    42000                           2500   
15    15         2000                    43000                           2500   
16    16         2000                    43000                           2500   
17    17         2000                    41000                           2500   
18    18         2000                    42000                           2500   
19    19         2000                    45000                           2500   
20    20         2000                    45000                           2500   
21    21         2000                    44000                           2500   
22    22         2000                    46000                           2500   
23    23         2000                    47000                           2500   
24    24         2000                    48000                           2500   
25    25         2000                    50000                           2500   

   Net amount recd by Bid winner Amount returned to everyone in the group  
1                          37500                                      400  
2                          39500                                      320  
3                          42500                                      200  
4                          45500                                       80  
5                          37500                                      400  
6                          39500                                      320  
7                          40500                                      280  
8                          41500                                      240  
9                          37500                                      400  
10                         39500                                      320  
11                         42500                                      200  
12                         45500                                       80  
13                         37500                                      400  
14                         39500                                      320  
15                         40500                                      280  
16                         40500                                      280  
17                         38500                                      360  
18                         39500                                      320  
19                         42500                                      200  
20                         42500                                      200  
21                         41500                                      240  
22                         43500                                      160  
23                         44500                                      120  
24                         45500                                       8

In [31]:
df['Amount returned to everyone in the group'].unique()

array([400, 320, 200, 80, 280, 240, 360, 160, 120, 0], dtype=object)

In [32]:
df['Amount returned to everyone in the group'].value_counts()

320    5
200    4
400    4
280    3
80     3
240    2
160    1
120    1
360    1
0      1
Name: Amount returned to everyone in the group, dtype: int64

In [33]:
#I1,I5,I9,I12 person gain equal loss end of the mature chit fund
I1=I5=I9=I12=6300

In [34]:
r1=r5=r9=r12=-6.048